src/Zip.Shared/ZipEntry.Write.cs

```diff
@@ -2396,7 +2420,10 @@ internal void WriteSecurityMetadata(Stream outstream)
                 // verification" value for the entry.
 
                 int keystrength = GetKeyStrengthInBits(Encryption);
```

Comment:
This line can be moved inside the if block and then you can update the Trace line to use `_aesCrypto_forWrite._KeyStrengthInBits` instead of `keystrength`.